src/main/java/com/epam/ta/reportportal/ws/model/attribute/ItemAttributeResource.java

```diff
@@ -28,18 +28,23 @@
  */
 public class ItemAttributeResource implements Serializable {
 
-	@Size(max = MAX_ATTRIBUTE_LENGTH)
 	private String key;
 
 	@NotBlank
-	@Size(min = MIN_ITEM_ATTRIBUTE_VALUE_LENGTH, max = MAX_ATTRIBUTE_LENGTH)
+	@Size(min = MIN_ITEM_ATTRIBUTE_VALUE_LENGTH)
 	private String value;
 
 	public ItemAttributeResource() {
 	}
 
 	public ItemAttributeResource(String key, String value) {
+		if (key != null && key.length() > MAX_ATTRIBUTE_LENGTH) {
```

Comment:
Substring can confuse user if saved only part of information. As for me response with Validation message is more correct

Reply:
As discussed cut of string with length more then 512 will be done in service-api.
Also as discussed it would be done silently, without validation message for user